src/librustdoc/html/markdown/footnotes.rs

```diff
@@ -33,21 +35,28 @@ impl<'a, I: Iterator<Item = SpannedEvent<'a>>> Footnotes<'a, I> {
         Footnotes { inner: iter, footnotes: FxIndexMap::default(), existing_footnotes, start_id }
     }
 
-    fn get_entry(&mut self, key: &str) -> (&mut Vec<Event<'a>>, usize) {
+    fn get_entry(&mut self, key: &str) -> (&mut Vec<Event<'a>>, usize, &mut usize) {
         let new_id = self.footnotes.len() + 1 + self.start_id;
         let key = key.to_owned();
-        let FootnoteDef { content, id } =
-            self.footnotes.entry(key).or_insert(FootnoteDef { content: Vec::new(), id: new_id });
+        let FootnoteDef { content, id, num_refs } = self
+            .footnotes
+            .entry(key)
+            .or_insert(FootnoteDef { content: Vec::new(), id: new_id, num_refs: 0 });
         // Don't allow changing the ID of existing entries, but allow changing the contents.
-        (content, *id)
+        (content, *id, num_refs)
     }
 
     fn handle_footnote_reference(&mut self, reference: &CowStr<'a>) -> Event<'a> {
         // When we see a reference (to a footnote we may not know) the definition of,
         // reserve a number for it, and emit a link to that number.
-        let (_, id) = self.get_entry(reference);
+        let (_, id, num_refs) = self.get_entry(reference);
+        *num_refs += 1;
+        let fnref_suffix = {
+            let num_refs = *num_refs;
+            if num_refs <= 1 { "".to_owned() } else { format!("-{num_refs}") }
+        };
```

Comment:
```suggestion
        let fnref_suffix =
            if *num_refs <= 1 { "".to_owned() } else { format!("-{num_refs}") };
```

Reply:
Oh, I used `{x}` (when `x` is a reference) in my programs and I somehow ignored [implementations for references](https://doc.rust-lang.org/1.89.0/std/fmt/trait.Display.html#impl-Display-for-%26mut+T).  Will change.